src/libexpr/eval.cc

```diff
@@ -199,6 +191,10 @@ void initGC()
     /* Initialise the Boehm garbage collector. */
     GC_set_all_interior_pointers(0);
 
+    /* Values are using interrior pointers to pack the value type. */
+    GC_register_displacement(1);
+    GC_register_displacement(2);
```

Comment:
What's the effect of this in relation to `GC_set_all_interior_pointers(0)` (which was necessary to prevent some big leaks)?